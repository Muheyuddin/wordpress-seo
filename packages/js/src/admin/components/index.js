export { default as RouteErrorFallback } from "./route-error-fallback";
export { default as SidebarLayout } from "./sidebar-layout";
export { default as Topbar } from "./topbar";
