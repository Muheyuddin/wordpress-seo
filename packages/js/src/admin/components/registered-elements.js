import { Fragment } from "@wordpress/element";
import { useRootContext } from "@yoast/ui-library";

/**
 * @returns {JSX.Element} The element.
 */
const RegisteredElements = () => {
	const { elements } = useRootContext();

	return elements.map( ( { key, value } ) => <Fragment key={ key }>{ value }</Fragment> );
};

export default RegisteredElements;
