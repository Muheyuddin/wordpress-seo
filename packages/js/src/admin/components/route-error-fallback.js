import { useCallback } from "@wordpress/element";
import { __ } from "@wordpress/i18n";
import { Alert, Button, Title } from "@yoast/ui-library";
import PropTypes from "prop-types";

/**
 * @param {Object} error The error instance.
 * @returns {JSX.Element} The element.
 */
const RouteErrorFallback = ( { error } ) => {
	const handleRefreshClick = useCallback( () => window?.location?.reload(), [] );

	return (
		<div role="alert" className="yst-max-w-screen-sm yst-p-8 yst-space-y-4">
			<Title>{ __( "Something went wrong. An unexpected error occurred.", "wordpress-seo" ) }</Title>
			<p>{ __( "We're very sorry, but it seems like the following error has interrupted our application:", "wordpress-seo" ) }</p>
			<Alert variant="error">{ error?.message || "Undefined error message." }</Alert>
			<p>{ __( "Unfortunately, this means that all your unsaved changes will be lost. You can try and refresh this page to resolve the problem.", "wordpress-seo" ) }</p>
			<div className="yst-flex yst-gap-2">
				<Button onClick={ handleRefreshClick }>{ __( "Refresh this page", "wordpress-seo" ) }</Button>
			</div>
		</div>
	);
};

RouteErrorFallback.propTypes = {
	error: PropTypes.object.isRequired,
};

export default RouteErrorFallback;
