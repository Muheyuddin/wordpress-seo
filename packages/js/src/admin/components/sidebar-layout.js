import PropTypes from "prop-types";

/**
 * @param {JSX.Element} children The main content.
 * @param {JSX.Element} sidebar The sidebar content.
 * @returns {JSX.Element} The layout.
 */
const SidebarLayout = ( { children, sidebar } ) => (
	<div className="yst-p-4 min-[783px]:yst-p-8 yst-flex yst-gap-4">
		<aside className="yst-sidebar yst-sidebar-nav yst-shrink-0 yst-hidden min-[783px]:yst-block yst-pb-6 yst-bottom-0 yst-w-56">
			{ sidebar }
		</aside>
		<div className="yst-flex yst-grow yst-flex-wrap">
			<div className="yst-grow yst-space-y-6 yst-mb-8 xl:yst-mb-0">
				<main className="yst-rounded-lg yst-bg-white yst-shadow">
					{ children }
				</main>
			</div>
		</div>
	</div>
);

SidebarLayout.propTypes = {
	children: PropTypes.node.isRequired,
	sidebar: PropTypes.node.isRequired,
};

export default SidebarLayout;
