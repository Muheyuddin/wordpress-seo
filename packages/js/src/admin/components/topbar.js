import { __ } from "@wordpress/i18n";
import { TopbarNavigation, useSvgAria } from "@yoast/ui-library";
import classNames from "classnames";
import { Link, useLocation } from "react-router-dom";
import { YoastLogo } from "../../settings/components";
import { useSelectAdmin } from "../hooks";

/**
 * @returns {JSX.Element} The topbar.
 */
const Topbar = () => {
	const isPremium = useSelectAdmin( "selectFromShared", [], "isPremium", false );
	const links = useSelectAdmin( "selectLinksFromRoutes" );
	const svgAriaProps = useSvgAria();
	const { pathname } = useLocation();

	return (
		<TopbarNavigation activePath={ pathname }>
			<TopbarNavigation.Container>
				<TopbarNavigation.Topbar>
					<TopbarNavigation.Start>
						<TopbarNavigation.MobileMenuButton
							openButtonScreenReaderText={ __( "Open top bar navigation menu", "wordpress-seo" ) }
							closeButtonScreenReaderText={ __( "Close top bar navigation menu", "wordpress-seo" ) }
						/>
					</TopbarNavigation.Start>

					<TopbarNavigation.Center>
						<TopbarNavigation.LogoContainer>
							<Link
								id="link-yoast-logo-topbar"
								to="/"
								className={ classNames(
									"yst-block yst-py-2.5 yst-border-b-2 yst-border-transparent focus:yst-ring-primary-500",
									pathname === "/" && "yst-border-primary-500"
								) }
								aria-label={ isPremium ? "Yoast SEO Premium" : "Yoast SEO" }
							>
								<YoastLogo className="yst-w-40" { ...svgAriaProps } />
							</Link>
						</TopbarNavigation.LogoContainer>
						<TopbarNavigation.LinksContainer>
							{ links.map( ( { children, ...props }, index ) => (
								<TopbarNavigation.Link
									key={ `link-${ index }` }
									as={ Link }
									hrefProp="to"
									{ ...props }
								>
									{ children }
								</TopbarNavigation.Link>
							) ) }
						</TopbarNavigation.LinksContainer>
					</TopbarNavigation.Center>
				</TopbarNavigation.Topbar>

				<TopbarNavigation.MobileContainer>
					{ links.map( ( { children, ...props }, index ) => (
						<TopbarNavigation.MobileLink
							key={ `link-mobile-${ index }` }
							as={ Link }
							hrefProp="to"
							{ ...props }
						>
							{ children }
						</TopbarNavigation.MobileLink>
					) ) }
				</TopbarNavigation.MobileContainer>
			</TopbarNavigation.Container>
		</TopbarNavigation>
	);
};

export default Topbar;
