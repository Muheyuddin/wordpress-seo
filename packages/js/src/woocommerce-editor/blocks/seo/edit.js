// import { useWooBlockProps } from "@woocommerce/block-templates";
import { useBlockProps } from "@wordpress/block-editor";
import { Slot } from "@wordpress/components";
// import { useEntityProp } from "@wordpress/core-data";
import { LocationProvider, Root } from "@yoast/externals/contexts";
import sortComponentsByRenderPriority from "../../../helpers/sortComponentsByRenderPriority";
import { SLOTS } from "../../constants";

const ROOT_CONTEXT = { locationContext: "product-seo" };

/**
 * The edit function describes the structure of your block in the context of the
 * editor. This represents what the editor will render when the block is used.
 *
 * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#edit
 *
 * @returns {JSX.Element} The element.
 */
export const Edit = ( { attributes, context: { postType } } ) => {
	/**
	 * React hook that is used to mark the block wrapper element.
	 * It provides all the necessary props like the class name.
	 *
	 * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-block-editor/#useblockprops
	 */
	//	const blockProps = useWooBlockProps();
	const blockProps = useBlockProps();

	//	const validation = useValidation( "testing", async function validate( ...args ) {
	//		console.error( "validating!", ...args );
	//		return false;// "TEST ERROR";
	//	} );

	// The "context.postType" is there because of the usesContext in the block.json.
	console.log( { attributes, postType, blockProps } );

	//	const postType = useContext( PostTypeContext );
	//	const [ id ] = useEntityProp( "postType", postType, "id" );
	//	const [ sku ] = useEntityProp( "postType", postType, "sku" );
	//	// This has an internal check to also get/set meta_data.
	//	const [ sku2 ] = useProductEntityProp( "sku", { postType, fallbackValue: "" } );

	// Editor data:
	//	const { editedRecord } = useEntityRecord( "postType", postType, id );
	//	const [ content ] = useEntityProp( "postType", postType, "description" );
	//	const [ title ] = useEntityProp( "postType", postType, "name" );
	//	const [ excerptOnly ] = useEntityProp( "postType", postType, "short_description" );
	//	// Excerpt: get from the content when excerptOnly is empty.
	//	// Maybe some more logic needed here for the decodeURI and auto-draft slug?
	//	const [ slug ] = useEntityProp( "postType", postType, "slug" );
	//	const [ generatedSlug ] = useEntityProp( "postType", postType, "generated_slug" );
	//	const [ images ] = useEntityProp( "postType", postType, "images" );
	//	const snippetPreviewImageUrl = images?.[ 0 ]?.src || null;
	// Content image: get from the content.
	// Base url: there is a permalink_template in the record. WC has a getPermalinkParts selector, but doesn't seem to work?
	//	const PERMALINK_PRODUCT_REGEX = /%(?:postname|pagename)%/;
	//	const postName = product.slug || product.generated_slug;
	//	const [ prefix, suffix ] = product.permalink_template.split( PERMALINK_PRODUCT_REGEX );

	//	const [ metaFocusKeyphrase, setMetaFocusKeyphrase ] = useProductEntityProp( "meta_data._yoast_wpseo_focuskw", { postType, fallbackValue: ""
	// } );

	// NOTE: the block.json NEEDS to have an attribute or this will not be interactive.
	return <div { ...blockProps }>
		<LocationProvider value="sidebar">
			<Root context={ ROOT_CONTEXT }>
				<Slot name={ SLOTS.seo }>
					{ ( fills ) => sortComponentsByRenderPriority( fills ) }
				</Slot>
			</Root>
		</LocationProvider>
	</div>;
};
