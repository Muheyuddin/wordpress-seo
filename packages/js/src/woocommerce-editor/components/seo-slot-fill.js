import { createSlotFill } from "@wordpress/components";
import sortComponentsByRenderPriority from "../../helpers/sortComponentsByRenderPriority";

const SLOT_FILL_NAME = "yoast-seo/woocommerce-editor/seo";

const { Slot, Fill } = createSlotFill( SLOT_FILL_NAME );

/**
 * @returns {JSX.Element} The element.
 */
export const SeoSlot = () => (
	<Slot>
		{ ( fills ) => sortComponentsByRenderPriority( fills ) }
	</Slot>
);

export const SeoFill = Fill;
