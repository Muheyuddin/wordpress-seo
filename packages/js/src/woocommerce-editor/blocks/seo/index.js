import { registerProductEditorBlockType } from "@woocommerce/product-editor";
import block from "./block.json";
import { Edit } from "./edit";

const { name, ...metadata } = block;

/**
 * Registers the SEO block to the product editor.
 *
 * @link https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/utils/register-product-editor-block-type.ts
 * @link https://developer.wordpress.org/block-editor/developers/block-api/#registering-a-block
 */
registerProductEditorBlockType( {
	name,
	metadata,
	settings: {
		example: {},
		edit: Edit,
	},
} );
