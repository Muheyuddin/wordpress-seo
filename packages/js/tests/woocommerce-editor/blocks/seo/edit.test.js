import { SlotFillProvider } from "@wordpress/components";
import { LocationConsumer, RootContext } from "@yoast/externals/contexts";
import { Edit } from "../../../../src/woocommerce-editor/blocks/seo/edit";
import { SeoFill } from "../../../../src/woocommerce-editor/components/seo-slot-fill";
import { render, screen } from "../../../test-utils";

jest.mock( "@woocommerce/block-templates", () => ( {
	useWooBlockProps: arg => arg,
} ) );

describe( "Edit", () => {
	it( "renders", () => {
		render( <Edit attributes={ { "data-testid": 1 } } context={ { postType: "product" } } /> );

		expect( screen.getByTestId( 1 ) ).toMatchSnapshot();
	} );

	it( "has a slot", () => {
		render( <SlotFillProvider>
			<Edit attributes={ { "data-testid": 1 } } context={ { postType: "product" } } />
			<SeoFill>Foo</SeoFill>
		</SlotFillProvider> );

		expect( screen.getByText( "Foo" ) ).toBeInTheDocument();
		expect( screen.getByTestId( 1 ) ).toMatchSnapshot();
	} );

	it( "provides the location", () => {
		render( <SlotFillProvider>
			<Edit attributes={ { "data-testid": 1 } } context={ { postType: "product" } } />
			<SeoFill>
				<LocationConsumer>
					{ ( location ) => location }
				</LocationConsumer>
			</SeoFill>
		</SlotFillProvider> );

		expect( screen.getByText( "sidebar" ) ).toBeInTheDocument();
		expect( screen.getByTestId( 1 ) ).toMatchSnapshot();
	} );

	it( "provides the locationContext", () => {
		render( <SlotFillProvider>
			<Edit attributes={ { "data-testid": 1 } } context={ { postType: "product" } } />
			<SeoFill>
				<RootContext.Consumer>
					{ ( { locationContext } ) => locationContext }
				</RootContext.Consumer>
			</SeoFill>
		</SlotFillProvider> );

		expect( screen.getByText( "product-seo" ) ).toBeInTheDocument();
		expect( screen.getByTestId( 1 ) ).toMatchSnapshot();
	} );

	it( "provides the postType", () => {
		render( <SlotFillProvider>
			<Edit attributes={ {} } context={ { postType: "product" } } />
			<SeoFill>
				<RootContext.Consumer>
					{ ( { postType } ) => postType }
				</RootContext.Consumer>
			</SeoFill>
		</SlotFillProvider> );

		expect( screen.getByText( "product" ) ).toBeInTheDocument();
	} );
} );
