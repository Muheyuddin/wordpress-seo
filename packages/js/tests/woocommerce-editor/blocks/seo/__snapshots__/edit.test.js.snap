// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit has a slot 1`] = `
<div
  data-testid="1"
>
  Foo
</div>
`;

exports[`Edit provides the location 1`] = `
<div
  data-testid="1"
>
  sidebar
</div>
`;

exports[`Edit provides the locationContext 1`] = `
<div
  data-testid="1"
>
  product-seo
</div>
`;

exports[`Edit renders 1`] = `
<div
  data-testid="1"
/>
`;
